### ISSUE TITLE
[java] Mark constant-only interfaces / classes that are instantiable

### ISSUE THREAD

**Issue by jsotuyod** [pmd/pmd#559]
Equivalent rule to [`java-design/UseUtilityClass`](https://pmd.github.io/pmd-5.8.1/pmd-java/rules/java/design.html#UseUtilityClass) but for classes that just have public constants.

These types have no behavior and should not be instantiable.

If the class / interface extends another, no warning should be added (behavior may be present on parent class / interface).

**Code Sample demonstrating the issue:**

```java
public class Constants {
    public static final String MAILING_FROM = "noreply@mycompany.com";
    public static final String USER_AGENT = "mycompany/spider 1.0"
}
```

```java
public interface Constants {
    public static final String MAILING_FROM = "noreply@mycompany.com";
    public static final String USER_AGENT = "mycompany/spider 1.0"
}
```

**Comment by oowekyala** [comment]
Really interesting! We could do that painlessly using signature matching ([cf metrics](https://github.com/oowekyala/pmd/wiki/Overview-of-the-implementation#operation-and-field-signatures)). It could even allow us to look upwards in the hierarchy to see whether the superclasses / -interfaces define some behaviour or not

I'm working on making signature matching available to rules and not just metrics atm. I hope it can be available by the end of the week

**Comment by jennyBu** [comment]
I think about implementing this rule. Is it important to do it similar to the ConstantsInInterface rule which is done the xpath way or could I also do the java approach?

**Comment by jsotuyod** [comment]
@jennyBu welcomed! All PRs are highly appreciated. There is no constraint as to how to build this, the rule should be achievable using both Java and XPath expressions, so whichever you are more comfortable with should be ok.

**Comment by jsotuyod** [comment]
@jennyBu I've noticed you already seem to have branches on your fork with both, this rule and #595. Is there anything missing? Any reason for not submitting those PRs? Let us know if we can help you in any way.
